python311Packages.autobahn: fix tests

dotlambda opened the pull request:
## Description of changes
cc https://github.com/NixOS/nixpkgs/pull/289734

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda committed:
python311Packages.pytest-asyncio_0_21: init at 0.21.1

Version 0.23 was accidentally a breaking change:
https://github.com/pytest-dev/pytest-asyncio/issues/706

dotlambda committed:
python311Packages.autobahn: use pytest-asyncio_0_21

With version 0.23 the tests fail with

    RuntimeError: Event loop is closed

mweinelt commented:
Fails on 3.12 due to spake build failure, if you want to tackle that as well.

dotlambda committed:
python312Packages.spake2: fix build

dotlambda committed:
python311Packages.py-ecc: 6.0.0 -> 7.0.0

Diff: https://github.com/ethereum/py_ecc/compare/v6.0.0...v7.0.0

Changelog: https://github.com/ethereum/py_ecc/blob/v7.0.0/CHANGELOG.rst

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -11600,6 +11600,16 @@ self: super: with self; {
 
   pytest-asyncio = callPackage ../development/python-modules/pytest-asyncio { };
 
+  pytest-asyncio_0_21 = pytest-asyncio.overridePythonAttrs (old: rec {
```
Do we expect this to be used by more than autobahn? if not we might as well make it local for autobahn since it is only used in checkInputs

dotlambda replied:
We might have to. The upstream issue makes it sound like autobahn isn't the only reverse dependency affected.

dotlambda committed:
python312Packages.eth-utils: 2.1.1 -> 4.0.0

Diff: https://github.com/ethereum/eth-utils/compare/v2.1.1...v4.0.0

Changelog: https://github.com/ethereum/eth-utils/blob/v4.0.0/docs/release_notes.rst

dotlambda committed:
python312Packages.eth-typing: 3.2.0 -> 4.0.0

Diff: https://github.com/ethereum/eth-typing/compare/refs/tags/v3.2.0...v4.0.0

Changelog: https://github.com/ethereum/eth-typing/blob/v4.0.0/docs/release_notes.rst

dotlambda committed:
python312Packages.py-ecc: disable failing tests
